ethcore/src/engines/clique/step_service.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright 2015-2019 Parity Technologies (UK) Ltd.
+// This file is part of Parity Ethereum.
+
+// Parity Ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
+
+use engines::Engine;
+use parity_machine::Machine;
+use std::sync::Weak;
+use std::time::Duration;
+use std::thread;
+use std::sync::Arc;
+use parking_lot::RwLock;
+
+pub struct StepService {
+	shutdown: Arc<RwLock<bool>>,
```

Comment:
Use `Arc<AtomicBool>` infavor of `Arc<RwLock<bool>>`

Reply:
But AtomicBool is very verbose, RwLock api is much nicer...

Reply:
But it is much faster

Reply:
Are we gonna change this, or leave it as is?

Reply:
https://github.com/goerli/parity-goerli/pull/83